API.md

```diff
@@ -503,36 +783,132 @@ frisby.create('First test')
 ```
 
 ### only()
+
 Exclusively run this test. When `toss()` is invoked, the test is wrapped in a
 Mocha `describe.only` block instead of a `describe` block.
 
+### timeout(ms)
+
+Sets the timeout for this request.
+
+| Parameter | Description | Required |
+| --------- | ----------- | -------- |
+| ms | Integer. Sets the mocha timeout for this request. | Yes |
+
+```javascript
+frisby.create('Long-running request')
+    .get('http://example.com/slowthing')
+    .timeout(5000)
+    .expectStatus(200)
+    .toss()
+```
+
+This function can also be called with no parameter to return the current configured timeout (either by default, by global setup or having used this function with a parameter previously).
+
+This is different from [expectMaxResponseTime](#expectmaxresponsetimems) in that expectMaxResponseTime will await the request to complete then test the amount of time that the request took (after any preceeding expect functions). This will abort the request and so not run any developer-defined expect functions on the response.
+
+### retry(count, backoff)
+
+Set the number of (and additional backoff between) retries for this test. Each retry will be the configured timeout plus (retry number x backoff) apart.
+
+| Parameter | Description | Required |
+| --------- | ----------- | -------- |
+| count | Integer. Number of times to retry (meaning that 0 is still a single attempt) | Yes |
+| backoff | Integer. Number of milliseconds to add to the wait between each successive retry. Defaults to 1000ms. | No |
+
+```javascript
+frisby.create('Get a flaky thing')
+    .get('http://example.com/thing-that-sometimes-responds')
+    .expectStatus(200)
+    .retry(2, 250)
+    .toss()
+```
+
+In this above example, the first request would wait 2000ms for a response (the default), then the first retry would begin 250ms later, allow another 2000ms for this to perhaps complete, then wait 500ms (2 x 250ms) before beginning the second retry.
+
+### baseUri(uri)
+
+Set the root URI/URL that will be prepended to every request, replacing anything set by `request.baseUri` in global setup.
+
+```javascript
+frisby.create('Simple Get')
+  .baseUri('http://httpbin.org')
+  .get('/get?foo=bar')
+  .expectStatus(200)
+.toss()
+```
+
+### waits(ms)
+
+Sets a period of time in milliseconds to wait after `toss()` before the request is sent. This can allow time for server-side processing between chained requests.
```

Comment:
The wait is between the start of the test and when the request is sent. The call to `toss()` could have occurred earlier. The wait happens after the `before()` hooks.

Reply:
Thanks! Fixed!